pallets/msa/src/benchmarking.rs

```diff
@@ -700,6 +817,351 @@ mod benchmarks {
+
+		let application_name = (1..s as u8).collect::<Vec<_>>();
+		let cid = "bafkreidgvpkjawlxz6sffxzwgooowe5yt7i6wsyg236mfoks77nywkptdq"
+			.as_bytes()
+			.to_vec();
+
+		let mut localized_names = BoundedBTreeMap::new();
+		let mut localized_cids = BoundedBTreeMap::new();
+
+		for i in 0..n {
+			let lang_code = make_lang_code(i as usize, lang_size as usize);
+			let lang = BoundedVec::try_from(lang_code).unwrap();
+			let name = BoundedVec::try_from(application_name.clone()).unwrap_or_default();
+			localized_names.try_insert(lang.clone(), name).unwrap();
+		}
+		for i in 0..m {
+			let lang_code = make_lang_code(i as usize, lang_size as usize);
+			let lang = BoundedVec::try_from(lang_code).unwrap();
+			let logo = BoundedVec::try_from(cid.clone()).unwrap();
+			localized_cids.try_insert(lang, logo).unwrap();
+		}
+
+		let provider_caller: T::AccountId = whitelisted_caller();
+		let (provider_id, provider_public_key) =
+			Msa::<T>::create_account(provider_caller.clone(), EMPTY_FUNCTION).unwrap();
+		let entry = ProviderRegistryEntry::default();
+
+		assert_ok!(Msa::<T>::create_provider_v2(
+			RawOrigin::Signed(provider_caller.clone()).into(),
+			entry
+		));
+		let application_payload = ApplicationContext {
+			default_name: BoundedVec::try_from(application_name).unwrap_or_default(),
+			localized_names,
+			default_logo_250_100_png_cid: BoundedVec::try_from(cid).unwrap(),
+			localized_logo_250_100_png_cids: localized_cids,
+		};
+
+		#[extrinsic_call]
+		_(RawOrigin::Root, provider_public_key, application_payload);
+
+		assert_eq!(NextApplicationIndex::<T>::get(ProviderId(provider_id)), 1);
+		assert!(ProviderToApplicationRegistry::<T>::get(ProviderId(provider_id), 0).is_some());
+		Ok(())
+	}
+
+	#[benchmark]
+	fn upload_logo() -> Result<(), BenchmarkError> {
+		let max_logo_size = T::MaxLogoSize::get();
+		let max_logo_bytes = vec![0u8; max_logo_size as usize];
+		let logo_cid = compute_cid(&max_logo_bytes);
+		let provider_caller: T::AccountId = whitelisted_caller();
+		let (_, provider_public_key) =
+			Msa::<T>::create_account(provider_caller.clone(), EMPTY_FUNCTION).unwrap();
+		let entry = ProviderRegistryEntry::default();
+
+		assert_ok!(Msa::<T>::create_provider_v2(
+			RawOrigin::Signed(provider_caller.clone()).into(),
+			entry
+		));
+
+		let input_bounded_cid = BoundedVec::try_from(logo_cid).unwrap();
+		let input_bounded_logo = BoundedVec::try_from(max_logo_bytes).unwrap();
+		ApprovedLogos::<T>::insert(&input_bounded_cid, BoundedVec::new());
+
+		#[extrinsic_call]
+		_(
+			RawOrigin::Signed(provider_public_key),
+			input_bounded_cid.clone(),
+			input_bounded_logo.clone(),
+		);
+
+		assert!(ApprovedLogos::<T>::get(input_bounded_cid.clone()).is_some());
+		let stored_logo_bytes = ApprovedLogos::<T>::get(&input_bounded_cid).unwrap();
+		assert_eq!(stored_logo_bytes, input_bounded_logo);
+		Ok(())
+	}
+
+	#[benchmark]
+	fn propose_to_update_provider() -> Result<(), BenchmarkError> {
```

Comment:
Benchmark for this function depends on two things

1. payload items have direct effect on the storage writes (logo adds)
2. stored state have direct effect on the storage writes (logo removals) (this should be maxed in benchmark and based on the actual value the rest should be refunded in extrinsic)